### PR TITLE
Feature/3 specflowplugin

### LINKED ISSUE #3: Implement SpecFlow plugin

### PR DESCRIPTION
**Pull request by pavelanihimovsky**
Resolves #3

### CODE REVIEW

File: AqualityTracking.Integrations/src/AqualityTracking.Integrations.Core/AqualityTrackingLifecycle.cs [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+
+        public void AddAttachment(string filePath)
+        {
+            if (Enabled)
+            {
+                testResultEndpoints.AddAttachment((int)currentTestResult.Value.Id, filePath);
+            }            
+        }
+
+        public void FinishTestExecution(FinalResultId finalResultId, string failReson)
```
Comment (mialeska) [inline AqualityTracking.Integrations/src/AqualityTracking.Integrations.Core/AqualityTrackingLifecycle.cs:100]: failReson -> failReason

File: AqualityTracking.Integrations/src/AqualityTracking.Integrations.Core/AqualityTrackingLifecycle.cs [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+        private static void SetCurrentTestRun(TestRun testRun)
+        {
+            currentTestRun = testRun;
+        }
+
+        public void StartTestExecution(string testName = null)
+        {
+            if (Enabled)
+            {
+                var currentTestName = testName ?? currentTest.Value.Name;
```
Comment (mialeska) [inline AqualityTracking.Integrations/src/AqualityTracking.Integrations.Core/AqualityTrackingLifecycle.cs:84]: when the currentTest is created initially? Seems that currentTest.Value could be not ititialized at this point, isn't it?

Comment (pavelanihimovsky) [reply AqualityTracking.Integrations/src/AqualityTracking.Integrations.Core/AqualityTrackingLifecycle.cs:84]: I set it in binding
```
[BeforeScenario(Order = int.MinValue)]
        public void SetCurrentTestData()
        {
            var currentTestCase = new Test
            {
                Name = $"{featureContext.FeatureInfo.Title}: {scenarioContext.ScenarioInfo.Title}"
            };
            lifecycle.SetCurrentTest(currentTestCase);
        }
```

Comment (pavelanihimovsky) [reply AqualityTracking.Integrations/src/AqualityTracking.Integrations.Core/AqualityTrackingLifecycle.cs]: I've updated lifecycle and method SetTest.

**Pull request merged**